chpl/chpl-service/src/main/java/gov/healthit/chpl/app/resource/G3Sed2015ResourceCreatorApp.java

```diff
@@ -0,0 +1,132 @@
+package gov.healthit.chpl.app.resource;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import org.apache.commons.lang.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+import org.springframework.context.support.AbstractApplicationContext;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
+
+import gov.healthit.chpl.app.AppConfig;
+import gov.healthit.chpl.app.presenter.CertifiedProductCsvPresenter;
+import gov.healthit.chpl.dao.EntityRetrievalException;
+import gov.healthit.chpl.domain.CertificationCriterion;
+import gov.healthit.chpl.domain.CertificationResult;
+import gov.healthit.chpl.domain.SEDRow;
+import gov.healthit.chpl.domain.TestParticipant;
+import gov.healthit.chpl.domain.TestTask;
+import gov.healthit.chpl.dto.CertificationCriterionDTO;
+import gov.healthit.chpl.dto.CertificationResultDetailsDTO;
+import gov.healthit.chpl.dto.CertificationResultTestTaskDTO;
+import gov.healthit.chpl.dto.CertifiedProductDetailsDTO;
+import gov.healthit.chpl.util.CertificationResultRules;
+@Transactional
+@Component("G3Sed2015ResourceCreatorApp")
+public class G3Sed2015ResourceCreatorApp extends SEDDownloadableResourceCreatorApp{
+	private static final String CRITERIA_NAME = "170.315 (g)(3)";
+    private static final String EDITION = "2015";
+    private static final Logger LOGGER = LogManager.getLogger(G3Sed2015ResourceCreatorApp.class);
+    
+    public G3Sed2015ResourceCreatorApp() {
+    	super();
+    }
+
+    /**
+     * This application generates a weekly summary email with an attached CSV
+     * providing CHPL statistics
+     * 
+     * @param args
+     * @throws Exception
+     */
+    public static void main(String[] args) throws Exception {
+        G3Sed2015ResourceCreatorApp app = new G3Sed2015ResourceCreatorApp();
+        app.setLocalContext();
+        AbstractApplicationContext context = new AnnotationConfigApplicationContext(AppConfig.class);
+        app.initiateSpringBeans(context);
+        app.runJob(args);
+        context.close();
+    }
+    
+    protected ArrayList<SEDRow> getRelevantListings() throws EntityRetrievalException {
+    	ArrayList<SEDRow> sedData = new ArrayList<SEDRow>();
+    	CertificationCriterionDTO certCrit = getCriteriaDao().getByNameAndYear(CRITERIA_NAME, EDITION);
+    	List<Long> ids = getCertificationResultDao().getCpIdsByCriterionId(certCrit.getId());
```

Comment:
FYI I believe you could have used the existing search code here instead of writing something of your own. We are able to send back search results (including the cp id) for a variety of search parameters. The CertifiedProductSearchResultDAOImpl search method takes a SearchRequest parameter that can contain criteria names.